docs/proposals/20191106-machine-load-balancer.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Machine Load Balancer 
+authors:
+  - "@moshloop"
+reviewers:
+    - "@akutz"
+    - "@ncdc"
+    - "@timothysc"
+    - "@vincepri"
+creation-date: 2019-11-06
+last-updated: 2019-11-06
+status: provisional
+---
+
+# Machine Load Balancer
+
+### Motivation
+
+* On-Premises clusters (without access to a Public Cloud Load Balancer as a Service) need a pluggable mechanism to enable the deployment of highly-available (HA) control planes using Cluster API (CAPI).
+
+* Multi-cluster environments need a mechanism to load balance ingress across multiple clusters in a safe and deterministic manner to support clusters as cattle / immutable clusters
+
+* Some environments need the ability to create different load balancing topologies e.g. 1 LB for control plane traffic, and another for external traffic to the kubernetes API. Or 1 LB per AZ to reduce inter-AZ traffic.
+
+#### Goals
+
+* Provide an extension point(s) to enable integrating external load balancers.
+* Allow the reuse of an LBP across CAPI providers, ex. AWS Elastic Load Balancers (ELB), F5, Netscaler, etc.
+* Represent the status of load balancer in resource status e.g. 4 of 5 members healthy.
+
+#### Non-Goals
+
+* Providing load balancers for non-control plane workloads i.e Service Type Load Balancer
+* Handling other Infrastructure Providers like networking and firewalls, This CAEP may, however, be merged into a larger CAEP when it is ready.
+
+## Summary
+
+The Machine Load Balancer (MLB) provides an abstraction for implementing load balancing across machines while also allowing the resulting  load balancer address to be used by any CRD.
+
+#### Coupling and Dependencies
+
+* The MLB depends on the [template operator](https://docs.google.com/document/d/1dJdnMozsWJ2ccPXWWkZpFVccFFSI7sYaJ6D25sZFlTU/edit#heading=h.vclg2zerysbp) to plumb the load balancer endpoint from the load balancer CRD, to the cluster CRD
```

Comment:
There should either be an additional comma after _plumb the load balancer endpoint_ **or** the command after _from the load balancer CRD_ should be dropped as the phrase _from the load balancer CRD_ is an interjection.